package/MDAnalysis/coordinates/OpenMM.py

```diff
@@ -0,0 +1,199 @@
+
+Example
+-------
+OpenMM can read various file formats into OpenMM objects.
+MDAnalysis can then convert some of these OpenMM objects into MDAnalysis Universe objects.
+
+    >>> import simtk.openmm.app as app
+    >>> import MDAnalysis as mda
+    >>> from MDAnalysis.tests.datafiles import PDBX
+    >>> pdbxfile = app.PDBxFile(PDBX)
+    >>> mda.Universe(pdbxfile)
+    <Universe with 60 atoms>
+
+
+
+Classes
+-------
+
+.. autoclass:: OpenMMSimulationReader
+   :members:
+
+.. autoclass:: OpenMMAppReader
+   :members:
+
+
+"""
+
+
+import numpy as np
+
+from . import base
+from .. import units
+
+
+class OpenMMSimulationReader(base.SingleFrameReaderBase):
+    """Reader for OpenMM Simulation objects
+
+
+    .. versionadded:: 2.0.0
+    """
+
+    format = "OPENMMSIMULATION"
+    units = {"time": "ps", "length": "nm", "velocity": "nm/ps", 
+            "force": "kJ/(mol*nm)", "energy": "kJ/mol"}
+
+    @staticmethod
+    def _format_hint(thing):
+        """Can this reader read *thing*?
+        """
+        try:
+            from simtk.openmm.app import Simulation
+        except ImportError:
+            return False
+        else:
+            return isinstance(thing, Simulation)
+
+    def _read_first_frame(self):
+        self.n_atoms = self.filename.topology.getNumAtoms()
+
+        self.ts = self._mda_timestep_from_omm_context()
+
+        if self.convert_units:
+            self.convert_pos_from_native(self.ts._pos)
+            self.ts.triclinic_dimensions = self.convert_pos_from_native(
+                self.ts.triclinic_dimensions, inplace=False
+            )
+            self.convert_velocities_from_native(self.ts._velocities)
+            self.convert_forces_from_native(self.ts._forces)
+            self.convert_time_from_native(self.ts.dt)
+
+    def _mda_timestep_from_omm_context(self):
+        """ Construct Timestep object from Openmm context """
+
+        state = self.filename.context.getState(-1, getVelocities=True, 
+                getForces=True, getEnergy=True)
+
+        n_atoms = self.filename.context.getSystem().getNumParticles()
+
+        ts = self._Timestep(n_atoms, **self._ts_kwargs)
+        ts.frame = 0
+        ts.data["time"] = state.getTime()._value
+        ts.data["potential_energy"] = self.convert_energies_from_native(
+                state.getPotentialEnergy()._value)
+        ts.data["kinetic_energy"] = self.convert_energies_from_native(
+                state.getKineticEnergy()._value)
+        ts.triclinic_dimensions = state.getPeriodicBoxVectors(
+                asNumpy=True)._value
+        ts.positions = state.getPositions(asNumpy=True)._value
+        ts.velocities = state.getVelocities(asNumpy=True)._value
+        ts.forces = state.getForces(asNumpy=True)._value
+
+        return ts
+
+    def convert_energies_from_native(self, energy):
```

Comment:
For this and the other conversion functions -- doesn't `getPotentialEnergy()` and the like return the quantity with units already attached (like simtk.physical.Quantity and simtk.physical.Unit)? Couldn't you just convert from that to the MDA unit instead?

Reply:
Yes it's true the getters return simtk.Quantity objects, but I wasn't sure what the correct syntax would be to map from the simtk Unit to floats in a correct MDAnalysis manner. If preferred, we can do the unit conversion solely with the simtk unit API

Reply:
Thanks, I think that's a bit more foolproof than `_value`.